src/IdentityServer/Licensing/v2/LicenseAccessor.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Duende Software. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using Duende.IdentityServer.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Duende.IdentityServer.Licensing.v2;
+
+/// <summary>
+/// Loads the license from configuration or a file, and validates its contents.
+/// </summary>
+internal class LicenseAccessor(IdentityServerOptions options, ILogger<LicenseAccessor> logger) : ILicenseAccessor
```

Comment:
Test this class against real licenses (commit expired licenses to repo and mock clock as needed)